haystack/preview/ready_made_pipelines/rag.py

```diff
@@ -0,0 +1,97 @@
+from typing import Optional, List
+
+from haystack.preview import Pipeline
+from haystack.preview.dataclasses import Answer
+from haystack.preview.document_stores import InMemoryDocumentStore
+from haystack.preview.components.retrievers import InMemoryBM25Retriever, InMemoryEmbeddingRetriever
+from haystack.preview.components.embedders import SentenceTransformersTextEmbedder
+from haystack.preview.components.generators import GPTGenerator
+from haystack.preview.components.builders.answer_builder import AnswerBuilder
+from haystack.preview.components.builders.prompt_builder import PromptBuilder
+
+
+class RAGPipeline:
+    """
+    A simple ready-made pipeline for RAG. It requires a populated document store.
+
+    If an embedding model is given, it uses embedding retrieval. Otherwise it falls back to BM25 retrieval.
+
+    Example usage:
+
+    ```python
+    rag_pipe = RAGPipeline(document_store=InMemoryDocumentStore())
+    answers = rag_pipe.run(query="Who lives in Rome?")
+    >>> Answer(data="Giorgio")
+    ```
+
+    """
+
+    def __init__(
+        self,
+        document_store: InMemoryDocumentStore,
```

Comment:
If we cannot customize the pipeline on the docstore I would exclude it from the init

Reply:
We need a populated document store to be able to retrieve documents from somewhere...